Source/WarpX.cpp

```diff
@@ -504,6 +504,10 @@ WarpX::ReadParameters ()
         std::vector<int> numprocs_in;
         utils::parser::queryArrWithParser(
             pp_warpx, "numprocs", numprocs_in, 0, AMREX_SPACEDIM);
+        #ifdef WARPX_DIM_RZ
+                WARPX_ALWAYS_ASSERT_WITH_MESSAGE( numprocs_in[0] == 1,
+                        "Domain decomposition in RZ works only along z direction");
+        #endif
 
         if (not numprocs_in.empty()) {
```

Comment:
Thank you, @oshapoval! I think the assert needs to go inside the `if` block below, because it needs to be checked only if `numprocs` is provided in the inputs (and since it is an optional parameter, that's not always the case). This should fix CI.
```suggestion

        if (not numprocs_in.empty()) {
#ifdef WARPX_DIM_RZ
            WARPX_ALWAYS_ASSERT_WITH_MESSAGE(numprocs_in[0] == 1,
                "Domain decomposition in RZ works only along z direction");
#endif
```